backend/authentication/user.py

```diff
@@ -34,16 +41,36 @@ def discord_mention(self) -> str:
     def decoded_token(self) -> dict[str, any]:
         return jwt.decode(self.token, SECRET_KEY, algorithms=["HS256"])
 
-    async def fetch_admin_status(self, request: Request) -> bool:
-        self.admin = await request.state.db.admins.find_one(
+    async def get_user_roles(self, database: Database) -> list[str]:
+        """Get a list of the user's discord roles."""
+        if not self.member:
+            return []
+
+        server_roles = await discord.get_roles(database)
+        roles = [role.name for role in server_roles if role.id in self.member.roles]
+
+        if "admin" in roles:
+            # Protect against collision with the forms admin role
+            roles.remove("admin")
+            roles.append("discord admin")
```

Comment:
Since the role name in the server is `Admins`, wouldn't there be no collision? Or am I misunderstanding something?

Reply:
No you're right, this is more of a precautionary measure though. Form admins have had a lot of perms, so reducing the surface for attack on that seems smart.